#Домашнее задание №2

##Задача №1
Считать данные с помощью panda. Вывести на экран первые 5 строк. Посмотреть на описание признаков и на их содержание

In [15]:
import pandas as pd
import numpy as np

In [16]:
df = pd.read_csv('./kc_house_data.csv')
df

id             date     price  bedrooms  bathrooms  \
0      7129300520  20141013T000000  221900.0         3       1.00   
1      6414100192  20141209T000000  538000.0         3       2.25   
2      5631500400  20150225T000000  180000.0         2       1.00   
3      2487200875  20141209T000000  604000.0         4       3.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
0             1180      5650     1.0           0     0  ...      7   
1             2570      7242     2.0           0     0  ...      7   
2              770     10000     1.0           0     0  ...      6   
3             1960      5000     1.0           0     0  ...      7   
4             1680      8080     1.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
3            1050            910      1965             0    98136  47.5208   
4            1680              0      1987             0    98074  47.6168   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[21613 rows x 21 columns]

In [17]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

##Задача №2
Проведите первичный анализ данных. Изучите типы данных. Найдите количество пропущенных ячеек в данных. Посчитайте основные статистики по всем признакам и поизучайте их. Пишите выводы.

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

### Ответ: Таблица не содердит пропусков. Состав: вещественные значения float64(5), целочисленные значения int64(15), строковые значения object(1).

##Задача №3
### Ответ
* 3.1 В каком диапазоне изменяются стоимости недвижимости? 
Стоимость изменяется в столбце price
* 3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?
Средняя 1986.552492, отклонение 685.391304
* 3.3 Как много домов с разными этажами в данных?
Ответ:
1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
* 3.4 Насколько хорошие состояния у домов в данных?
50% домов старше 1975 и не имеют реновации. Жилой фонд достаточно старый.
* 3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?
Самый новый дом построен в 2015, самый старый дом в 1900

In [19]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [20]:
df['yr_built'].min()

1900

In [24]:
df['yr_built'].max()

2015

In [31]:
df['floors'].value_counts()

1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: floors, dtype: int64

In [27]:
print(df.describe(include=['object']))

                   date
count             21613
unique              372
top     20140623T000000
freq                142


## Задача №4
### Ответ:
* 4.1 Сколько в среднем стоят дома, у которых 2 спальни?
Стоимость составляет 401372.681884058
* 4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?
Средняя площадь 20442.524776214832
* 4.3 Как много домов коснулся ремонт?
Из общего колличества домов 21613 единиц, ремонт сделан только в 914

* 4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?
Стоимость домов в с оценкой выше 10 в среднем составляет 1678635.12, а с оценкой меньше 4 = 189750.0

In [47]:
df[df['bedrooms'] == 2]['price'].mean()

401372.681884058

In [48]:
df[df['price'] > 600000]['sqft_lot'].mean()

20442.524776214832

In [50]:
df[df['grade'] > 10]['price'].mean()

1678635.1175298805

In [51]:
df[df['grade'] < 4]['price'].mean()

189750.0

In [32]:
df['yr_renovated'].value_counts()

0       20699
2014       91
2013       37
2003       36
2005       35
        ...  
1951        1
1959        1
1948        1
1954        1
1944        1
Name: yr_renovated, Length: 70, dtype: int64

## Задача №5
### Ответ:
* 5.1 Выберите дом клиенту  Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?
homeone
* 5.2 Выберите дом клиенту Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?
hometwo  
Средняя стоимость такой квартиры будет 1233906.695
* 5.3 Выберите дом клиенту Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?
Средняя стоимость такой квартиры будет 131928.57142857142
homethree


In [53]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [55]:
homeone = df[ (df ['bathrooms'] >= 3) & (df ['view'] > 0)]
homeone

id             date      price  bedrooms  bathrooms  \
15     9297300055  20150124T000000   650000.0         4       3.00   
58     7922800400  20140827T000000   951000.0         5       3.25   
153    7855801670  20150401T000000  2250000.0         4       3.25   
185     825059270  20141121T000000  1095000.0         5       3.00   
216      46100204  20150221T000000  1505000.0         5       3.00   
...           ...              ...        ...       ...        ...   
21521  7430500110  20141209T000000  1378000.0         5       3.50   
21540   715010530  20150113T000000  1881580.0         5       3.50   
21576  9253900271  20150107T000000  3567000.0         5       4.50   
21579  7011201004  20140529T000000   645000.0         3       3.25   
21593  8672200110  20150317T000000  1088000.0         5       3.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
15            2950      5000     2.0           0     3  ...      9   
58            3250     14342     2.0           0     4  ...      8   
153           5180     19850     2.0           0     3  ...     12   
185           4090     12850     1.0           0     2  ...     10   
216           3300     33474     1.0           0     3  ...      9   
...            ...       ...     ...         ...   ...  ...    ...   
21521         5150     12230     2.0           0     2  ...     10   
21540         4410     13000     2.0           0     3  ...     10   
21576         4850     10584     2.0           1     4  ...     10   
21579         1730      1229     2.0           0     2  ...      9   
21593         4170      8142     2.0           0     2  ...     10   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
15           1980            970      1979             0    98126  47.5714   
58           3250              0      1968             0    98008  47.5880   
153          3540           1640      2006             0    98006  47.5620   
185          2090           2000      1986             0    98033  47.6627   
216          1870           1430      1957          1991    98040  47.5673   
...           ...            ...       ...           ...      ...      ...   
21521        3700           1450      2007             0    98008  47.6249   
21540        2920           1490      2014             0    98006  47.5382   
21576        3540           1310      2007             0    98008  47.5943   
21579        1320            410      2008             0    98119  47.6374   
21593        4170              0      2006             0    98056  47.5354   

          long  sqft_living15  sqft_lot15  
15    -122.375           2140        4000  
58    -122.116           2960       11044  
153   -122.162           3160        9750  
185   -122.188           2540       10270  
216   -122.210           3836       20953  
...        ...            ...         ...  
21521 -122.090           2940       13462  
21540 -122.111           5790       12969  
21576 -122.110           3470       18270  
21579 -122.369           1710        1686  
21593 -122.181           3030        7980  

[637 rows x 21 columns]

In [56]:
hometwo = df[(df ['sqft_above'] >= 1980) & (df ['view'] > 0)]['price'].mean()
hometwo

1233906.6952288218

In [62]:
homethree = df[(df ['sqft_basement'] == 0) & (df ['floors'] == 2)& (df ['price'] <= 150000)]['price'].mean()
homethree

131928.57142857142